From ff111d6f91550a6c64c41c5bc0b1ee0db11d528b Mon Sep 17 00:00:00 2001
From: Tulio Beloqui <tulio@pexip.com>
Date: Thu, 23 Nov 2023 09:06:55 +0100
Subject: [PATCH 3/9] cairo-win32-font: added calls to CAIRO_MUTEX_INITIALIZE
 to initialize the static mutex, as these are missing for the static version

In the dynamic version, these are called from cairo-win32-system whenever
the dll is loaded.
---
 src/win32/cairo-win32-font.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/win32/cairo-win32-font.c b/src/win32/cairo-win32-font.c
index a561e74a4..b8a18fb4e 100644
--- a/src/win32/cairo-win32-font.c
+++ b/src/win32/cairo-win32-font.c
@@ -160,6 +160,7 @@ _get_global_font_dc (void)
     HDC hdc;
 
     if (!hdc_tls_index) {
+    CAIRO_MUTEX_INITIALIZE ();
 	CAIRO_MUTEX_LOCK (_cairo_win32_font_dc_mutex);
 	if (!hdc_tls_index) {
 	    hdc_tls_index = TlsAlloc ();
@@ -1685,6 +1686,7 @@ _cairo_win32_font_face_hash_table_destroy (void)
 static cairo_hash_table_t *
 _cairo_win32_font_face_hash_table_lock (void)
 {
+	CAIRO_MUTEX_INITIALIZE ();
     CAIRO_MUTEX_LOCK (_cairo_win32_font_face_mutex);
 
     if (unlikely (cairo_win32_font_face_hash_table == NULL))
-- 
2.46.0

